spec/system/admin/order_spec.rb

```diff
@@ -126,6 +126,73 @@ def new_order_with_distribution(distributor, order_cycle)
     expect(page).not_to have_select2 "add_variant_id", with_options: [product.name]
   end
 
+  context "deleting item of an order" do
+    context "when there a more than one items in the order" do
+      let(:line_item) { create(:line_item) }
+
+      before do
+        order.line_items << line_item
+        login_as_admin_and_visit spree.edit_admin_order_path(order)
+        find("a.delete-item").click
+        expect(page).to have_content "Are you sure?"        
+      end
+
+      it "show a modal 'Are you sure?' that the user can close and then nothing change" do
+        within(".modal", visible: true) do
+          expect do
+            click_on("Cancel")
+            expect(page).not_to have_content "Are you sure?"
+          end.not_to change { order.line_items.length }
+        end
+      end
+
+      it "show a modal 'Are you sure?' that the user confirm and then the item is deleted" do
+        expect(order.line_items.length).to eq(2)
+        within(".modal", visible: true) do
+          expect do
+            click_on("OK")
+            expect(page).not_to have_css(".modal", visible: true)
+          end.to change { order.reload.line_items.length }.by(-1)
+        end
+      end
+    end
+
+    context "when it is the last item of an order" do
+      before do
+        # specify that order has only one line item
+        order.line_items = [order.line_items.first]
+        login_as_admin_and_visit spree.edit_admin_order_path(order)
+        find("a.delete-item").click 
+        within(".modal", visible: true) do
+          # ignore first modal by confirming it
+          click_on("OK")
+        end
+      end
+
+      context "it shows a second modal about last item deletion and therefore about order cancellation" do
+        
+        it "that the user can close and then nothing change" do
+          expect(page).to have_content "This will cancel the current order."
+          within(".modal", visible: true) do
+            click_on("Cancel")
+          end
+          
+          expect(order.reload.line_items.length).to eq(1)
+          expect(page).to have_selector('tr.stock-item', count: 1)
+        end
+
+        it "that the user can confirm and then the order is cancelled" do
```

Comment:
@mkllnk I wanted to test here if the email is well send to the consumer (or not). Could you point me any advice/tips on how to realize it in a context of a spec?

Reply:
Search the code base for the email helper `performing_deliveries` and you'll find some examples.
https://github.com/openfoodfoundation/openfoodnetwork/blob/04e2b8728b06fc28744d17bf7ddfeb454555afdd/spec/controllers/user_confirmations_controller_spec.rb#L72-L78

Reply:
There's also the `setup_email` helper which you may need. I'm not sure. It may be obsolete thanks to some work Matt did recently.

Reply:
I've tested via `have_enqueued_mail(Spree::OrderMailer, :cancel_email)` thanks for pointing me ;)

Then I discovered that an email is still sent even if we unselect the checkbox (thanks to the test that fails...), I must know understand why it send an email ;)

Reply:
On commit ed9710bc0